packages/xstate-react/src/createSelector.ts

```diff
@@ -0,0 +1,12 @@
+import { StateFrom, StateMachine } from 'xstate';
+
+/**
+ * Allows you to specify a typesafe selector based on a machine definition.
+ *
+ * @example
+ * const getIsToggedOn = createSelector(machine, state => state.matches('toggledOn'));
+ */
+export const createSelector = <TMachine extends StateMachine<any, any, any>, T>(
+  _machine: TMachine,
+  selector: (state: StateFrom<TMachine>) => T
+) => selector;
```

Comment:
I wonder if it wouldnt be more ergonomic to embed `useSelector` directly in this:
```suggestion
import { StateMachine, ActorRefFrom } from 'xstate';
import { useSelector } from './useSelector';

/**
 * Allows you to specify a typesafe `useSelector` based on a machine definition.
 *
 * @example
 * const useIsToggedOn = createUseSelector(machine, state => state.matches('toggledOn'));
 */
export const createUseSelector = <TMachine extends StateMachine<any, any, any>, T>(
  _machine: TMachine,
  selector: (state: TMachine['initialState']) => T
) => (actorRef: ActorRefFrom<TMachine>) => useSelector(actorRef, selector);
```

Reply:
Especially that this way it makes more sense for this to live in the `@xstate/react`. The previous version had no dependency on React whatsoever 🤔

Reply:
@Andarist I have thoughts about this! This is worse because you sometimes want to write selectors which depend on other selectors, and compose them. `createSelector` is a pattern common in Redux, and could also be used with `reselect` if you fancied it.

Reply:
Ok, I see that point but them it would IMHO make sense to move this to the `xstate` package itself - since this might be useful for all of the "integration" packages like `@xstate/react`, `@xstate/vue`, etc. Or maybe to a separate package that those integration packages would reexport?

Reply:
Yep, seems like it'll be useful in all packages. I think it'll be useful enough (and small enough) that it can fit into core, but also happy to put it into a separate package.

Reply:
Put it in core.